### PR TITLE
feat(ui): Add `priority` prop to DropdownMenuItemV2 

### PR DESCRIPTION
**Pull request by vuluongj20**
**Without `priority: 'danger'`:**
Screen Shot 2022-02-08 at 2 55 21 PM

**With `priority: 'danger'`**
Screen Shot 2022-02-08 at 2 55 41 PM

### CODE REVIEW

**Review by davidenwang — APPROVED** [review]

File: static/app/components/dropdownMenuItemV2.tsx [hunk 1]
```diff
@@ -37,6 +33,11 @@ export type MenuItemProps = {
    * when `children` is also defined.
    */
   isSubmenu?: boolean;
+  /**
+   * Item label. Should prefereably be a string. If not, make sure that
+   * there are appropriate aria-labels.
```
Comment (priscilawebdev) [inline static/app/components/dropdownMenuItemV2.tsx:38]: 💜

File: static/app/components/dropdownMenuItemV2.tsx [hunk 2]
```diff
@@ -248,16 +264,26 @@ const MenuItem = withRouter(
 );
 export default MenuItem;
 
-const MenuItemWrap = styled('li')<{isDisabled?: boolean}>`
+const MenuItemWrap = styled('li')<{
+  isDisabled?: boolean;
+  isFocused?: boolean;
+  priority?: 'primary' | 'danger';
```
Comment (priscilawebdev) [inline static/app/components/dropdownMenuItemV2.tsx:270]: should we create a` type Priority = 'primary' | 'danger'` and reuse it here and in the props?

File: static/app/components/dropdownMenuItemV2.tsx [hunk 3]
```diff
@@ -267,14 +293,30 @@ const MenuItemWrap = styled('li')<{isDisabled?: boolean}>`
   }
 `;
 
-const InnerWrap = styled('div')<{isFocused: boolean}>`
+const getHoverBackground = (theme: Theme, priority?: 'primary' | 'danger') => {
```
Comment (priscilawebdev) [inline static/app/components/dropdownMenuItemV2.tsx:296]: if we create the type, we could reuse here too ^^

File: static/app/components/dropdownMenuItemV2.tsx [hunk 4]
```diff
@@ -320,21 +362,23 @@ const LabelWrap = styled('div')`
...
   margin-bottom: 0;
   line-height: 1.4;
   white-space: nowrap;
   ${overflowEllipsis}
-
-  ${p => p.isDisabled && `color: ${p.theme.subText};`}
 `;
 
-const Details = styled('p')`
+const Details = styled('p')<{isDisabled: boolean; priority?: 'primary' | 'danger'}>`
```
Comment (priscilawebdev) [inline static/app/components/dropdownMenuItemV2.tsx:372]: and here too

**Review by priscilawebdev — APPROVED** [review]
left some suggestions but it looks good to me

File: static/app/components/dropdownMenuItemV2.tsx (hunk 2 above)
Comment (vuluongj20) [reply static/app/components/dropdownMenuItemV2.tsx]: oh great point! done!

**Pull request merged**